posthog/models/element/element.py

```diff
@@ -57,7 +58,15 @@ def elements_to_string(elements: list[Element]) -> str:
     return ";".join(ret)
 
 
+@lru_cache(maxsize=10000)
 def chain_to_elements(chain: str) -> list[Element]:
```

Comment:
this is called in multiple places and is entirely deterministic so we may as well cache it here too